sosw/components/config.py

```diff
@@ -45,6 +45,95 @@
 from sosw.components.dynamo_db import DynamoDbClient
 
 
+class SecretsManager:
+
+    secretsmanager_client = None
+
+
+    def __init__(self, test=False, **kwargs):
+
+        self.test = test
+
+        if not self.test:
+            self.test = True if os.environ.get('STAGE') == 'test' or os.environ.get('autotest') == 'True' else False
+
+
+    def _get_secretsmanager_client(self):
+
+        if self.secretsmanager_client is None:
+            self.secretsmanager_client = boto3.client('secretsmanager')
+
+        return self.secretsmanager_client
+
+
+    def call_boto_secrets_with_pagination(self, f, **kwargs):
+        """
+        Invoke SecretsManager functions with the ability to paginate results.
+
+        :param str f:           SecretsManager function to invoke.
+        :param object kwargs:   Keyword arguments for the function to invoke.
+        :rtype list
+        :return:                List of paginated responses.
+        """
+
+        secretsmanager_client = self._get_secretsmanager_client()
+
+        func = getattr(secretsmanager_client, f)
+        can_paginate = getattr(secretsmanager_client, 'can_paginate')(f)
+
+        if can_paginate:
+            logging.debug('SecretsManager.{%} can natively paginate', f)
+            paginator = secretsmanager_client.get_paginator(f)
+            response = paginator.paginate(**kwargs)
+            return response
+
+        else:
+            logging.debug('SecretsManager.{%} can not natively paginate', f)
+            response_list = []
+            response = func(**kwargs)
+            response_list.append(response)
+            while 'NextToken' in response:
+                kwargs['NextToken'] = response['NextToken']
+                response_list.append(func(**kwargs))
+            return response_list
+
+
+    def get_secrets_credentials(self, **kwargs):
+        """
+        Retrieve the credentials with given name or tag from AWS SecretsManager and return as a dictionary.
+        :param  kwargs:  {type: tag/name, value: value_name}
+        :rtype:             dict
+        :return:            Some credentials
+        """
+
+        filters, secrets_dict = [], {}
+        filter_type, value = kwargs.get('type'), kwargs.get('value')
+        valid_types = ['tag', 'name']
+
+        if not filter_type or filter_type not in valid_types:
+            raise KeyError('Error no type Tag/Name provided')
+
+        if not value:
+            raise KeyError('Error no value provided')
+
+        filters = [{'Key': 'name', 'Values': [value]}] if filter_type == 'name' else \
+            [{'Key': 'tag-value', 'Values': [value]}]
+
+        secretsmanager_client = boto3.client('secretsmanager')
```

Comment:
Use your getter please. No need to re-instantiate this client

Reply:
fixed